akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1120,13 +1146,22 @@ class DDataShardCoordinator(
         context.become(waitingForUpdate(evt, afterUpdateCallback, newRemainingKeys))
 
     case UpdateTimeout(CoordinatorStateKey, Some(`evt`)) =>
-      log.error(
-        "The ShardCoordinator was unable to update a distributed state within 'updating-state-timeout': {} millis (retrying). " +
-        "Perhaps the ShardRegion has not started on all active nodes yet? event={}",
-        writeMajority.timeout.toMillis,
-        evt)
-      // repeat until UpdateSuccess
-      sendCoordinatorStateUpdate(evt)
+      if (terminated) {
+        log.error(
+          "The ShardCoordinator was unable to update a distributed state within 'updating-state-timeout': {} millis (terminating). " +
+          "Perhaps the ShardRegion has not started on all active nodes yet? event={}",
+          writeMajority.timeout.toMillis,
+          evt)
+        onTerminated()
```

Comment:
I wonder if this is the right thing to do for the UpdateTimeout? Comparing `onTerminated` with `unbecomeAfterUpdate` the main difference is that the `afterUpdateCallback` isn't called, but it kind of continues with stashed messages anyway until it reaches the stashed `Terminate`.

Maybe a safer (more like it was before) option would be to `stop` here? Meaning that it will try to complete pending updates, but if UpdateTimeout it doesn't continue with more.